tests/lib.rs

```diff
@@ -166,427 +166,427 @@ use std::path::PathBuf;
+#[case::pl_1_2_1_2dcode_raw_6_json("PL/1.2.1/2DCode/raw/6.json")]
+#[case::pl_1_2_1_2dcode_raw_7_json("PL/1.2.1/2DCode/raw/7.json")]
+#[case::pl_1_2_1_2dcode_raw_8_json("PL/1.2.1/2DCode/raw/8.json")]
+#[case::pl_1_2_1_2dcode_raw_9_json("PL/1.2.1/2DCode/raw/9.json")]
+#[case::pl_1_3_0_2dcode_raw_1_json("PL/1.3.0/2DCode/raw/1.json")]
+#[case::pl_1_3_0_2dcode_raw_10_json("PL/1.3.0/2DCode/raw/10.json")]
+#[case::pl_1_3_0_2dcode_raw_11_json("PL/1.3.0/2DCode/raw/11.json")]
+#[case::pl_1_3_0_2dcode_raw_12_json("PL/1.3.0/2DCode/raw/12.json")]
+#[case::pl_1_3_0_2dcode_raw_13_json("PL/1.3.0/2DCode/raw/13.json")]
+#[case::pl_1_3_0_2dcode_raw_14_json("PL/1.3.0/2DCode/raw/14.json")]
+#[case::pl_1_3_0_2dcode_raw_2_json("PL/1.3.0/2DCode/raw/2.json")]
+#[case::pl_1_3_0_2dcode_raw_3_json("PL/1.3.0/2DCode/raw/3.json")]
+#[case::pl_1_3_0_2dcode_raw_4_json("PL/1.3.0/2DCode/raw/4.json")]
+#[case::pl_1_3_0_2dcode_raw_5_json("PL/1.3.0/2DCode/raw/5.json")]
+#[case::pl_1_3_0_2dcode_raw_6_json("PL/1.3.0/2DCode/raw/6.json")]
+#[case::pl_1_3_0_2dcode_raw_7_json("PL/1.3.0/2DCode/raw/7.json")]
+#[case::pl_1_3_0_2dcode_raw_8_json("PL/1.3.0/2DCode/raw/8.json")]
+#[case::pl_1_3_0_2dcode_raw_9_json("PL/1.3.0/2DCode/raw/9.json")]
+#[case::pt_1_0_0_2dcode_raw_1_json("PT/1.0.0/2DCode/raw/1.json")]
+#[case::pt_1_0_0_2dcode_raw_2_json("PT/1.0.0/2DCode/raw/2.json")]
+#[case::pt_1_0_0_2dcode_raw_3_json("PT/1.0.0/2DCode/raw/3.json")]
+#[case::pt_1_0_0_2dcode_raw_4_json("PT/1.0.0/2DCode/raw/4.json")]
+#[case::pt_1_0_0_2dcode_raw_5_json("PT/1.0.0/2DCode/raw/5.json")]
+#[case::pt_1_3_0_2dcode_raw_1_json("PT/1.3.0/2DCode/raw/1.json")]
+#[case::pt_1_3_0_2dcode_raw_2_json("PT/1.3.0/2DCode/raw/2.json")]
+#[case::pt_1_3_0_2dcode_raw_3_json("PT/1.3.0/2DCode/raw/3.json")]
+// #[case::pt_1_3_0_2dcode_raw_4_json("PT/1.3.0/2DCode/raw/4.json")] // SKIPPED. Weird difference in date field (`sc`). Expected "2021-06-30T12:34:56+00:00", found: "2021-06-30T12:34:56Z".
+#[case::pt_1_3_0_2dcode_raw_5_json("PT/1.3.0/2DCode/raw/5.json")]
+#[case::ro_2dcode_raw_1_json("RO/2DCode/raw/1.json")]
+#[case::ro_2dcode_raw_2_json("RO/2DCode/raw/2.json")]
+#[case::ro_2dcode_raw_3_json("RO/2DCode/raw/3.json")]
+#[case::ro_2dcode_raw_4_json("RO/2DCode/raw/4.json")]
+#[case::se_2dcode_raw_1_json("SE/2DCode/raw/1.json")]
+#[case::se_2dcode_raw_2_json("SE/2DCode/raw/2.json")]
+#[case::se_2dcode_raw_3_json("SE/2DCode/raw/3.json")]
+#[case::se_2dcode_raw_4_json("SE/2DCode/raw/4.json")]
+#[case::se_2dcode_raw_5_json("SE/2DCode/raw/5.json")]
+// #[case::sg_2dcode_raw_1_json("SG/2DCode/raw/1.json")] // SKIPPED. Name section does not have the `fn` field. Should we make that an `Option`?
+// #[case::sg_2dcode_raw_2_json("SG/2DCode/raw/2.json")] // SKIPPED. Name section does not have the `fn` field. Should we make that an `Option`?
+// #[case::sg_2dcode_raw_3_json("SG/2DCode/raw/3.json")] // SKIPPED. Name section does not have the `fn` field. Should we make that an `Option`?
```

Comment:
Do we have any hint from the specs if this is optional or not?

Reply:
Good one, let me try to find something about this... It feels like a flaky test to me (I don't believe it makes sense to have a certificate without family name), but it's certainly worth spending a bit more time on this.

Reply:
Apparently only fnt is required, so `fn` should be an `Option` :/

source: https://github.com/ehn-dcc-development/ehn-dcc-schema/blob/release/1.3.0/DCC.Core.Types.schema.json#L17-L63

I'll open a new issue for this!

Reply:
Opened #26